x-pack/solutions/security/packages/kbn-cloud-security-posture/graph/src/components/edge/default_edge.tsx

```diff
@@ -14,55 +14,68 @@ import { getMarkerStart, getMarkerEnd } from './markers';
 
 type EdgeColor = EdgeViewModel['color'];
 
-export function DefaultEdge({
-  id,
-  label,
-  sourceX,
-  sourceY,
-  sourcePosition,
-  targetX,
-  targetY,
-  targetPosition,
-  data,
-}: EdgeProps) {
-  const { euiTheme } = useEuiTheme();
-  const color: EdgeColor = data?.color ?? 'primary';
+const dashedStyle = {
+  strokeDasharray: '2 2',
+};
 
-  const [edgePath] = getSmoothStepPath({
-    // sourceX and targetX are adjusted to account for the shape handle position
-    sourceX: sourceX - getShapeHandlePosition(data?.sourceShape),
+export const DefaultEdge = memo(
+  ({
+    id,
+    label,
+    sourceX,
     sourceY,
     sourcePosition,
-    targetX: targetX + getShapeHandlePosition(data?.targetShape),
+    targetX,
     targetY,
     targetPosition,
-    borderRadius: 15,
-    offset: 0,
-  });
+    data,
+  }: EdgeProps) => {
+    const { euiTheme } = useEuiTheme();
+    const color: EdgeColor = data?.color ?? 'primary';
+    const sourceMargin = getShapeHandlePosition(data?.sourceShape);
+    const targetMargin = getShapeHandlePosition(data?.targetShape);
+    const markerStart =
+      data?.sourceShape !== 'label' && data?.sourceShape !== 'group'
+        ? getMarkerStart(color)
+        : undefined;
+    const markerEnd =
+      data?.targetShape !== 'label' && data?.targetShape !== 'group'
+        ? getMarkerEnd(color)
+        : undefined;
```

Comment:
nit, you can do something like this to simplify reading
```ts
// can probably name it better than just shapes
export const shapes = ['label', 'group'];

const markerStart = !shapes.includes(data?.sourceShape) ? getMarkerStart(color) : undefined;
const markerEnd = !shapes.includes(data?.targetShape) ? getMarkerEnd(color) : undefined;
```

Reply:
since the default behaviour when data is missing is to have a marker

For me, it looks less intuitive to read
```
    const markerStart =
      !data?.sourceShape || !NODES_WITHOUT_MARKER.includes(data?.sourceShape)
        ? getMarkerStart(color)
        : undefined;
    const markerEnd =
      !data?.sourceShape || !NODES_WITHOUT_MARKER.includes(data?.sourceShape)
        ? getMarkerEnd(color)
        : undefined;
```

wdyt?

Reply:
looks good as well!